controllers/transactionsController.js

```diff
@@ -0,0 +1,136 @@
+const Portfolio = require('../models').Portfolio;
+const League = require('../models').League;
+const Transaction = require('../models').Transaction;
+const User = require('../models').User;
+
+const get = require('../utils/request').getRequest;
+const formatDate = require('../utils/dates').formatDate;
+
+async function getStockData(key,symbol) {
+  let stockPrice;
+  try {
+    stockPrice = await get(`https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${symbol}&apikey=${key}`);
+  } catch (error) {
+    console.log(error)
+  }
+  
+  return stockPrice.data["Global Quote"]["05. price"];
+}
+
+module.exports = {
+  getTransactions(req,res) {
+    console.log("Reached")
+    const sessionID = 'abcde12345';
+    const leagueID = req.params.leagueID;
+    return User.findOne({where: {sessionID: sessionID}}).then(user => {
+      Portfolio.findOne({
+        where: {
+          leagueID: leagueID,
+          userID: user.id
+        }
+      }).then(portfolio => {
+        if (portfolio === null) {
+          res.status(400).send({message: "Transactions do not exist."});
+        } else {
+          Transaction.findAll({
+            where: {
+              portfolioID: portfolio.id
+            }
+          }).then(transactions => {
+            res.status(200).send(transactions)
+          }).catch(error => {
+            res.status(400).send(error)
+            console.log(error)
+          })
+        }
+      }).catch(error => {
+        res.status(400).send(error)
+        console.log(error)
+      })
+    }).catch(error => {
+      res.status(400).send(error)
+      console.log(error)
+    })
+  },
+  
+  makeTransaction(req,res) {
+    const leagueID = req.body.leagueID;
+    const sessionID = 'abcde12345';
+    // const sessionID = req.sessionId;
+    const API_KEY = 'U864TMWAO0GRH22S';
+    let transactionValue = 0;
+    const symbol = req.body.stockSymbol;
+    let price = req.body.price;
+    // get(`https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=${symbol}&apikey=${API_KEY}`).then(res => {
+    //   price = res.data["Global Quote"]["05. price"];
+    // }).catch(error => {
+    //   console.log(error);
+    //   res.status(400).send({message: "That is not a valid stock symbol."})
+    // });
+    // console.log("Symbol: ", symbol);
+    // console.log("Price: ", price);
+    
+    return User.findOne({where: {sessionID: sessionID}}).then(user => {
+
+      // Creates a new transaction in the database using information from the request body.
+      Transaction.create({
+        volume: req.body.volume,
+        type: req.body.type,
+        datetime: req.body.datetime,
+        price: price,
+        stockSymbol: symbol,
+        portfolioID: req.body.portfolioID
```

Comment:
you should get the portfolio first and then use the id of the portfolio instead of passing in a portfolio id in the request body